src/pages/components/Card.tsx

```diff
@@ -0,0 +1,51 @@
+import { useEffect, useState } from "react";
+import { GetProducts } from "../api/get-products"
+import Image from "next/image";
+
+interface Product {
+    name: string;
+    price: {
+        formattedValue: string;
+    };
+    imageUrl?: string;
+}
+
+export default function Card() {
+    const [products, setProducts] = useState<Product[]>([])
+
+    useEffect(() => {
+        const loadProducts = async () => {
+            const allProducts = await GetProducts();
+            const firstFiveProducts = allProducts.slice(0, 5);
+            setProducts(firstFiveProducts);
+        };
+
+        loadProducts();
+    }, []);
```

Comment:
Lembra que comentamos que a premissa principal do next é fazer requisições server side? Nesse caso, a chamada só vai ser feito quando o componente for montado, ou seja, pelo client side. Sugiro que dê uma olhada nessa documentação: https://nextjs.org/docs/pages/building-your-application/data-fetching/get-static-props